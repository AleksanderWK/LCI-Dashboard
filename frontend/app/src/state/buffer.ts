import {atom, atomFamily} from "recoil";
import {DataPoints} from "../pages/SessionView";

export const sessionCodesState = atom<string[]>({
    key: "sessionCodes",
    default: []
});

/*
 *  An atomFamily that stores buffer data for each session
 */
export const bufferState = atomFamily<{active: boolean; data: DataPoints[]}, string>({
    key: "buffer",
    default: {active: false, data: []}
});
