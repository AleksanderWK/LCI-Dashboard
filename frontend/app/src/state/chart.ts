import {atom} from "recoil";

/*
 *  An atom that stores the current callback function ID (increased by 1 for each new callback)
 */
export const callbackFunctionIdState = atom<number>({
    key: "callbackFunctionId",
    default: 0
});

interface CallbackFunctions {
    [key: number]: () => void;
}

/*
 *  An atom that stores all functions to be called on every chart update
 */
export const callbackFunctionsState = atom<CallbackFunctions>({
    key: "callbackFunctions",
    default: {}
});
