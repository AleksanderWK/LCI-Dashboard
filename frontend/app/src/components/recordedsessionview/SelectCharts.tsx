import React from "react";
import {createStyles, makeStyles, Theme} from "@material-ui/core/styles";
import {Button, List, ListItem, ListItemIcon, ListItemText, Checkbox, Typography} from "@material-ui/core";
import {useRecoilState, useSetRecoilState} from "recoil";
import {MMDVariables, Variable} from "../../constants";
import {selectChartsPopupOpenState} from "../../state/popup";
import {selectedRecordingActiveContainersState} from "../../state/recordedSession";

const useStyles = makeStyles((theme: Theme) =>
    createStyles({
        btn: {
            color: theme.palette.text.default
        },
        grid: {
            display: "grid",
            gridTemplateColumns: "1fr",
            justify: "space-between",
            gap: 18,
            color: theme.palette.text.default,
            width: "100%"
        },
        btnGroup: {
            display: "flex",
            width: "100%"
        },
        checkbox: {
            color: theme.palette.primary.main
        }
    })
);

/**
 * Handles selection of charts for the recorded session
 */
export default function SelectCharts(): JSX.Element {
    const classes = useStyles();
    const setPopupOpen = useSetRecoilState(selectChartsPopupOpenState);
    const [activeContainers, setActiveContainers] = useRecoilState(selectedRecordingActiveContainersState);

    /**
     * Adds or removes the given variable chart
     * @param {Variable} variable - The variable to be removed
     */
    const handleCheck = (variable: Variable) => () => {
        setActiveContainers((prevState) => {
            const list = [...prevState];
            if (list.includes(variable)) {
                list.splice(list.indexOf(variable), 1);
                return list;
            }
            list.push(variable);
            return list;
        });
    };

    /**
     * Adds or removes all of the charts
     * @param {boolean} checkAll - Whether all should be added (true) or removed (false)
     */
    const handleCheckAll = (checkAll: boolean) => {
        setActiveContainers((prevValue) => {
            return checkAll
                ? [
                      ...prevValue,
                      ...Object.values(Variable).filter(
                          (variable) => MMDVariables[variable].enabled && !prevValue.includes(variable)
                      )
                  ]
                : [];
        });
    };

    return (
        <div className={classes.grid}>
            <Typography variant="h1">Select Variables</Typography>
            <List style={{maxHeight: "400px", overflowY: "auto"}}>
                {Object.values(Variable)
                    // Variables that aren't enabled shouldn't be displayed
                    .filter((variable) => MMDVariables[variable].enabled)
                    .map((variable, index) => {
                        const name = MMDVariables[variable].name;
                        const labelId = `checkbox-list-label-${name}`;
                        // Generates checkbox and label for each variable
                        return (
                            <ListItem key={index} role={undefined} dense button onClick={handleCheck(variable)}>
                                <ListItemIcon className={classes.checkbox}>
                                    <Checkbox
                                        edge="start"
                                        checked={activeContainers.includes(variable)}
                                        tabIndex={-1}
                                        disableRipple
                                        color="primary"
                                        className={classes.checkbox}
                                        inputProps={{"aria-labelledby": labelId}}
                                    />
                                </ListItemIcon>
                                <ListItemText disableTypography id={labelId} primary={name} />
                            </ListItem>
                        );
                    })}
            </List>
            <div className={classes.btnGroup}>
                {
                    // Select all and remove all-buttons
                    Object.values(Variable).filter((variable) => MMDVariables[variable].enabled).length !=
                    activeContainers.length ? (
                        <Button
                            className={classes.btn}
                            onClick={() => {
                                handleCheckAll(true);
                            }}
                        >
                            <Checkbox
                                tabIndex={-1}
                                disableRipple
                                checked={false}
                                className={classes.checkbox}
                                color="primary"
                            ></Checkbox>
                            Select All
                        </Button>
                    ) : (
                        <Button
                            className={classes.btn}
                            onClick={() => {
                                handleCheckAll(false);
                            }}
                        >
                            <Checkbox
                                tabIndex={-1}
                                disableRipple
                                checked={true}
                                className={classes.checkbox}
                                color="primary"
                            ></Checkbox>
                            Remove All
                        </Button>
                    )
                }
                <Button
                    className={classes.btn}
                    style={{marginLeft: "auto"}}
                    onClick={() => {
                        setPopupOpen(false);
                    }}
                >
                    Done
                </Button>
            </div>
        </div>
    );
}
