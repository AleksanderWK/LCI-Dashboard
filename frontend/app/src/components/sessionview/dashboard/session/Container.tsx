import React, {useRef, useState} from "react";
import {MMDVariables, Variable} from "../../../../constants";
import Menu from "../Menu";
import LineChart from "../LineChart";
import Numeric from "../Numeric";
import ContainerCard from "../../../common/ContainerCard";
import Tooltip from "../../../common/Tooltip";
import {makeStyles, Theme, createStyles, IconButton, Typography} from "@material-ui/core";
import InfoOutlinedIcon from "@material-ui/icons/InfoOutlined";
import MoreVertIcon from "@material-ui/icons/MoreVert";
import CalculatingIndicator from "../CalculatingIndicator";
import {useRecoilState, useRecoilValue, useSetRecoilState} from "recoil";
import {selectedSessionDataLengthVariableState, selectedSessionIdState} from "../../../../state/session";
import XRangeChart from "../XRangeChart";
import {viewState, selectedSessionActiveContainersState, selectedSessionLayoutState} from "../../../../state/dashboard";

const useStyles = makeStyles((theme: Theme) =>
    createStyles({
        header: {
            display: "grid",
            gridTemplateColumns: "1fr 56px",
            gap: theme.spacing(2)
        },
        menu: {
            display: "grid",
            gridTemplateColumns: "24px 24px",
            gap: theme.spacing(1)
        },
        iconButton: {
            padding: 0
        },
        infoIcon: {
            cursor: "default"
        }
    })
);

interface Props {
    variable: Variable;
}

export default function Container(props: Props): JSX.Element {
    const classes = useStyles();

    const dataLength = useRecoilValue(selectedSessionDataLengthVariableState(props.variable));

    const selectedSessionId = useRecoilValue(selectedSessionIdState);
    const [view, setView] = useRecoilState(viewState([selectedSessionId, props.variable]));
    const setActiveContainers = useSetRecoilState(selectedSessionActiveContainersState);
    const setSelectedSessionLayout = useSetRecoilState(selectedSessionLayoutState);

    const [menuOpen, setMenuOpen] = useState<boolean>(false);

    const menuAnchorElement = useRef<HTMLDivElement | null>(null);

    function removeContainer(): void {
        setActiveContainers((prevState) => {
            const newState = [...prevState];
            return newState.filter((item) => item != props.variable);
        });

        // Remove from layout
        setSelectedSessionLayout((prevValue) => {
            const layout = [...prevValue];
            return layout.filter((item) => item.i != props.variable);
        });
    }

    return (
        <ContainerCard>
            <>
                <div className={classes.header}>
                    <Typography variant="h2" noWrap={true}>
                        {MMDVariables[props.variable].name}
                    </Typography>

                    <div className={`${classes.menu} noDrag`} ref={menuAnchorElement}>
                        <Tooltip variable={props.variable}>
                            <IconButton
                                aria-label="info"
                                disableFocusRipple={true}
                                disableRipple={true}
                                disableTouchRipple={true}
                                className={`${classes.iconButton} ${classes.infoIcon}`}
                            >
                                <InfoOutlinedIcon color="action" />
                            </IconButton>
                        </Tooltip>
                        <IconButton
                            aria-label="settings"
                            className={classes.iconButton}
                            onClick={() => setMenuOpen(true)}
                        >
                            <MoreVertIcon color="action" />
                        </IconButton>
                    </div>

                    <Menu
                        open={menuOpen}
                        anchorEl={menuAnchorElement.current}
                        isDetailedView={view === "chart"}
                        onShowMore={() => setView("chart")}
                        onShowLess={() => setView("numeric")}
                        onRemoveView={() => removeContainer()}
                        onMenuClose={() => setMenuOpen(false)}
                    />
                </div>
                {MMDVariables[props.variable].calculationTime && dataLength === 0 ? (
                    <CalculatingIndicator variable={props.variable} />
                ) : view === "chart" && props.variable != Variable.EducationalSpecificEmotions ? (
                    <LineChart variable={props.variable} />
                ) : view === "chart" ? (
                    <XRangeChart variable={props.variable} />
                ) : (
                    <Numeric variable={props.variable} />
                )}
            </>
        </ContainerCard>
    );
}
