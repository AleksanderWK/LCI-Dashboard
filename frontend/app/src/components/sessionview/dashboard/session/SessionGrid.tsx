import {createStyles, makeStyles} from "@material-ui/core";
import React, {useMemo} from "react";
import GridLayout, {Layout, WidthProvider} from "react-grid-layout";
import {useRecoilCallback, useRecoilState, useRecoilValue} from "recoil";
import {Variable} from "../../../../constants";
import {
    selectedSessionActiveContainersState,
    selectedSessionLayoutItemState,
    selectedSessionLayoutState
} from "../../../../state/dashboard";
import packer from "../../../../utils/packer";
import {ResizeIcon} from "../../../common/Icons";
import Container from "./Container";

const useStyles = makeStyles(() =>
    createStyles({
        resizeIcon: {
            position: "absolute",
            right: "4px",
            bottom: "4px",
            width: "10px",
            height: "10px",
            "&:hover": {
                cursor: "se-resize"
            }
        }
    })
);

const StaticGridLayout = WidthProvider(GridLayout);

/**
 * The grid layout for the live session view (a specific session).
 * Renders draggable and resizable containers.
 */
export default function SessionGrid(): JSX.Element {
    const classes = useStyles();

    const selectedSessionActiveContainers = useRecoilValue(selectedSessionActiveContainersState);

    const [selectedSessionLayout, setSelectedSessionLayout] = useRecoilState(selectedSessionLayoutState);

    /**
     * Get the layout from state (position, width and height) for a given variable
     */
    const getLayoutItem = useRecoilCallback(({snapshot}) => (variable: Variable): Layout | undefined => {
        return snapshot.getLoadable(selectedSessionLayoutItemState(variable)).getValue();
    });

    /**
     * Renders memoized containers for each selected variable
     * The containers are rerendered only if the selected containers is changed (one is added or removed)
     */
    const containers = useMemo(() => {
        const layout: Layout[] = [];
        return selectedSessionActiveContainers.map((variable) => {
            const existingItem = getLayoutItem(variable);
            const item = existingItem
                ? existingItem
                : {i: variable, w: 4, h: 3, minW: 2, minH: 3, ...packer(4, 3, layout, 12)};
            layout.push(item);
            return (
                <div key={variable} data-grid={item}>
                    <Container variable={variable} />
                </div>
            );
        });
    }, [selectedSessionActiveContainers]);

    return (
        <StaticGridLayout
            cols={12}
            rowHeight={60}
            containerPadding={[0, 0]}
            margin={[10, 10]}
            draggableCancel=".noDrag"
            isBounded
            resizeHandle={<ResizeIcon className={`${classes.resizeIcon} noDrag`} color="action" />}
            layout={selectedSessionLayout}
            onLayoutChange={(layout) => {
                // Check if the previous layout is different to the new one before updating state
                // This prevents unnecessary rerenders
                if (JSON.stringify(layout) !== JSON.stringify(selectedSessionLayout)) {
                    setSelectedSessionLayout(layout);
                }
            }}
        >
            {containers}
        </StaticGridLayout>
    );
}
