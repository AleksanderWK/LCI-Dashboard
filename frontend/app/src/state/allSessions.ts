/*
 *  State used in the all sessions view
 */

import {atom, selector} from "recoil";
import {Variable} from "../constants";
import {EducationalSpecificEmotions, sessionsState, sessionVariableDataState} from "./session";

export interface allsessionsObject {
    sessionId: number;
    studentName: string;
    data: [number, number | EducationalSpecificEmotions][];
}

/*
 *  An atom that stores which variable is selected for the all sessions view
 */
export const selectedAllSessionVariableState = atom<Variable | null>({
    key: "selectedAllSessionVariable",
    default: null
});

/*
 *  A selector that returns array of objects that each contain student name and session data for
 *  the currently selected variable.
 */
export const allSessionsState = selector<allsessionsObject[]>({
    key: "allSessions",
    get: ({get}) => {
        const result: allsessionsObject[] = [];
        const sessions = get(sessionsState);
        const selectedVariable = get(selectedAllSessionVariableState);
        sessions.forEach((session) => {
            if (selectedVariable != null) {
                const data = get(sessionVariableDataState([selectedVariable, session._id]));
                result.push({sessionId: session._id, studentName: session.student.name, data: data});
            }
        });
        return result;
    }
});
