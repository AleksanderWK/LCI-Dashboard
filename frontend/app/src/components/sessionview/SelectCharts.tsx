import React from "react";
import {createStyles, makeStyles, Theme} from "@material-ui/core/styles";
import List from "@material-ui/core/List";
import ListItem from "@material-ui/core/ListItem";
import ListItemIcon from "@material-ui/core/ListItemIcon";
import ListItemText from "@material-ui/core/ListItemText";
import Checkbox from "@material-ui/core/Checkbox";
import {selectedSessionActiveContainersState} from "../../state/session";
import {useRecoilState} from "recoil";
import {MMDVariables, Variable} from "../../constants";

const useStyles = makeStyles((theme: Theme) =>
    createStyles({
        root: {
            width: "100%",
            maxWidth: 360,
            backgroundColor: theme.palette.background.paper
        }
    })
);

export default function SelectCharts(): JSX.Element {
    const classes = useStyles();
    const [activeContainers, setActiveContainers] = useRecoilState(selectedSessionActiveContainersState);

    const handleToggle = (key: Variable) => () => {
        const newContainers = {...activeContainers};
        newContainers[key] = {...activeContainers[key], active: !activeContainers[key].active};
        setActiveContainers(newContainers);
    };

    return (
        <List className={classes.root}>
            {Object.values(Variable).map((variable, index) => {
                const name = MMDVariables[variable].name;
                const labelId = `checkbox-list-label-${name}`;
                return (
                    <ListItem key={index} role={undefined} dense button onClick={handleToggle(variable)}>
                        <ListItemIcon>
                            <Checkbox
                                edge="start"
                                checked={activeContainers[variable].active}
                                tabIndex={-1}
                                disableRipple
                                inputProps={{"aria-labelledby": labelId}}
                            />
                        </ListItemIcon>
                        <ListItemText id={labelId} primary={name} />
                    </ListItem>
                );
            })}
        </List>
    );
}
