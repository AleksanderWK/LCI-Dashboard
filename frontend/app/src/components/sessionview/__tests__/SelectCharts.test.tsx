import React from "react";
import ReactDOM from "react-dom";
import {render, screen} from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import {RecoilRoot} from "recoil";
import SelectCharts from "../SelectCharts";
import {act} from "react-dom/test-utils";
import {selectedSessionIdState} from "../../../state/session";

/*  Selected containers at start are changing frequently during development,
    snapshot testing isn't applicable'
it("popup matches snapshot", () => {
    const {baseElement} = render(
        <RecoilRoot>
            <SelectCharts />
        </RecoilRoot>
    );
    expect(baseElement).toMatchSnapshot();
});
*/

it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(
        <RecoilRoot>
            <SelectCharts />
        </RecoilRoot>,
        div
    );
});

it("select all button works properly", () => {
    act(() => {
        render(
            <RecoilRoot
                initializeState={(snap) => {
                    snap.set(selectedSessionIdState, 1);
                }}
            >
                <SelectCharts />
            </RecoilRoot>
        );
    });

    const selectAllBtn = screen.getByTestId("btn-select-all");

    expect(selectAllBtn).toBeInTheDocument();
    act(() => userEvent.click(selectAllBtn));
    expect(screen.getByTestId("btn-remove-all")).toBeInTheDocument();
    act(() => userEvent.click(screen.getByText("Cognitive Load")));
    expect(selectAllBtn).toBeInTheDocument();
});
