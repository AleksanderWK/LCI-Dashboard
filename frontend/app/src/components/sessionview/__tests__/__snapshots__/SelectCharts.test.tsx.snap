// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`popup matches snapshot 1`] = `
<body>
  <div>
    <div
      class="makeStyles-grid-2"
    >
      <h1
        class="MuiTypography-root MuiTypography-h1"
      >
        Select Variable Charts
      </h1>
      <ul
        class="MuiList-root MuiList-padding"
      >
        <div
          aria-disabled="false"
          class="MuiButtonBase-root MuiListItem-root MuiListItem-dense MuiListItem-gutters MuiListItem-button"
          tabindex="0"
        >
          <div
            class="MuiListItemIcon-root makeStyles-checkbox-5"
          >
            <span
              aria-disabled="false"
              class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-6 MuiCheckbox-root MuiCheckbox-colorPrimary makeStyles-checkbox-5 MuiIconButton-colorPrimary MuiIconButton-edgeStart"
            >
              <span
                class="MuiIconButton-label"
              >
                <input
                  aria-labelledby="checkbox-list-label-Cognitive Load"
                  class="PrivateSwitchBase-input-9"
                  data-indeterminate="false"
                  tabindex="-1"
                  type="checkbox"
                  value=""
                />
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                  />
                </svg>
              </span>
            </span>
          </div>
          <div
            class="MuiListItemText-root makeStyles-listItemText-4 MuiListItemText-dense"
            id="checkbox-list-label-Cognitive Load"
          >
            Cognitive Load
          </div>
          <span
            class="MuiTouchRipple-root"
          />
        </div>
        <div
          aria-disabled="false"
          class="MuiButtonBase-root MuiListItem-root MuiListItem-dense MuiListItem-gutters MuiListItem-button"
          tabindex="0"
        >
          <div
            class="MuiListItemIcon-root makeStyles-checkbox-5"
          >
            <span
              aria-disabled="false"
              class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-6 MuiCheckbox-root MuiCheckbox-colorPrimary makeStyles-checkbox-5 MuiIconButton-colorPrimary MuiIconButton-edgeStart"
            >
              <span
                class="MuiIconButton-label"
              >
                <input
                  aria-labelledby="checkbox-list-label-Perceived Difficulty"
                  class="PrivateSwitchBase-input-9"
                  data-indeterminate="false"
                  tabindex="-1"
                  type="checkbox"
                  value=""
                />
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                  />
                </svg>
              </span>
            </span>
          </div>
          <div
            class="MuiListItemText-root makeStyles-listItemText-4 MuiListItemText-dense"
            id="checkbox-list-label-Perceived Difficulty"
          >
            Perceived Difficulty
          </div>
          <span
            class="MuiTouchRipple-root"
          />
        </div>
        <div
          aria-disabled="false"
          class="MuiButtonBase-root MuiListItem-root MuiListItem-dense MuiListItem-gutters MuiListItem-button"
          tabindex="0"
        >
          <div
            class="MuiListItemIcon-root makeStyles-checkbox-5"
          >
            <span
              aria-disabled="false"
              class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-6 MuiCheckbox-root MuiCheckbox-colorPrimary makeStyles-checkbox-5 MuiIconButton-colorPrimary MuiIconButton-edgeStart"
            >
              <span
                class="MuiIconButton-label"
              >
                <input
                  aria-labelledby="checkbox-list-label-Familiarity"
                  class="PrivateSwitchBase-input-9"
                  data-indeterminate="false"
                  tabindex="-1"
                  type="checkbox"
                  value=""
                />
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                  />
                </svg>
              </span>
            </span>
          </div>
          <div
            class="MuiListItemText-root makeStyles-listItemText-4 MuiListItemText-dense"
            id="checkbox-list-label-Familiarity"
          >
            Familiarity
          </div>
          <span
            class="MuiTouchRipple-root"
          />
        </div>
        <div
          aria-disabled="false"
          class="MuiButtonBase-root MuiListItem-root MuiListItem-dense MuiListItem-gutters MuiListItem-button"
          tabindex="0"
        >
          <div
            class="MuiListItemIcon-root makeStyles-checkbox-5"
          >
            <span
              aria-disabled="false"
              class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-6 MuiCheckbox-root MuiCheckbox-colorPrimary makeStyles-checkbox-5 MuiIconButton-colorPrimary MuiIconButton-edgeStart"
            >
              <span
                class="MuiIconButton-label"
              >
                <input
                  aria-labelledby="checkbox-list-label-Information Processing Index"
                  class="PrivateSwitchBase-input-9"
                  data-indeterminate="false"
                  tabindex="-1"
                  type="checkbox"
                  value=""
                />
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                  />
                </svg>
              </span>
            </span>
          </div>
          <div
            class="MuiListItemText-root makeStyles-listItemText-4 MuiListItemText-dense"
            id="checkbox-list-label-Information Processing Index"
          >
            Information Processing Index
          </div>
          <span
            class="MuiTouchRipple-root"
          />
        </div>
        <div
          aria-disabled="false"
          class="MuiButtonBase-root MuiListItem-root MuiListItem-dense MuiListItem-gutters MuiListItem-button"
          tabindex="0"
        >
          <div
            class="MuiListItemIcon-root makeStyles-checkbox-5"
          >
            <span
              aria-disabled="false"
              class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-6 MuiCheckbox-root MuiCheckbox-colorPrimary makeStyles-checkbox-5 MuiIconButton-colorPrimary MuiIconButton-edgeStart"
            >
              <span
                class="MuiIconButton-label"
              >
                <input
                  aria-labelledby="checkbox-list-label-Physiological Arousal"
                  class="PrivateSwitchBase-input-9"
                  data-indeterminate="false"
                  tabindex="-1"
                  type="checkbox"
                  value=""
                />
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                  />
                </svg>
              </span>
            </span>
          </div>
          <div
            class="MuiListItemText-root makeStyles-listItemText-4 MuiListItemText-dense"
            id="checkbox-list-label-Physiological Arousal"
          >
            Physiological Arousal
          </div>
          <span
            class="MuiTouchRipple-root"
          />
        </div>
        <div
          aria-disabled="false"
          class="MuiButtonBase-root MuiListItem-root MuiListItem-dense MuiListItem-gutters MuiListItem-button"
          tabindex="0"
        >
          <div
            class="MuiListItemIcon-root makeStyles-checkbox-5"
          >
            <span
              aria-disabled="false"
              class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-6 MuiCheckbox-root MuiCheckbox-colorPrimary makeStyles-checkbox-5 MuiIconButton-colorPrimary MuiIconButton-edgeStart"
            >
              <span
                class="MuiIconButton-label"
              >
                <input
                  aria-labelledby="checkbox-list-label-Engagement"
                  class="PrivateSwitchBase-input-9"
                  data-indeterminate="false"
                  tabindex="-1"
                  type="checkbox"
                  value=""
                />
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                  />
                </svg>
              </span>
            </span>
          </div>
          <div
            class="MuiListItemText-root makeStyles-listItemText-4 MuiListItemText-dense"
            id="checkbox-list-label-Engagement"
          >
            Engagement
          </div>
          <span
            class="MuiTouchRipple-root"
          />
        </div>
        <div
          aria-disabled="false"
          class="MuiButtonBase-root MuiListItem-root MuiListItem-dense MuiListItem-gutters MuiListItem-button"
          tabindex="0"
        >
          <div
            class="MuiListItemIcon-root makeStyles-checkbox-5"
          >
            <span
              aria-disabled="false"
              class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-6 MuiCheckbox-root MuiCheckbox-colorPrimary makeStyles-checkbox-5 MuiIconButton-colorPrimary MuiIconButton-edgeStart"
            >
              <span
                class="MuiIconButton-label"
              >
                <input
                  aria-labelledby="checkbox-list-label-Physiological Stress"
                  class="PrivateSwitchBase-input-9"
                  data-indeterminate="false"
                  tabindex="-1"
                  type="checkbox"
                  value=""
                />
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                  />
                </svg>
              </span>
            </span>
          </div>
          <div
            class="MuiListItemText-root makeStyles-listItemText-4 MuiListItemText-dense"
            id="checkbox-list-label-Physiological Stress"
          >
            Physiological Stress
          </div>
          <span
            class="MuiTouchRipple-root"
          />
        </div>
        <div
          aria-disabled="false"
          class="MuiButtonBase-root MuiListItem-root MuiListItem-dense MuiListItem-gutters MuiListItem-button"
          tabindex="0"
        >
          <div
            class="MuiListItemIcon-root makeStyles-checkbox-5"
          >
            <span
              aria-disabled="false"
              class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-6 MuiCheckbox-root MuiCheckbox-colorPrimary makeStyles-checkbox-5 MuiIconButton-colorPrimary MuiIconButton-edgeStart"
            >
              <span
                class="MuiIconButton-label"
              >
                <input
                  aria-labelledby="checkbox-list-label-Emotional Regulation"
                  class="PrivateSwitchBase-input-9"
                  data-indeterminate="false"
                  tabindex="-1"
                  type="checkbox"
                  value=""
                />
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                  />
                </svg>
              </span>
            </span>
          </div>
          <div
            class="MuiListItemText-root makeStyles-listItemText-4 MuiListItemText-dense"
            id="checkbox-list-label-Emotional Regulation"
          >
            Emotional Regulation
          </div>
          <span
            class="MuiTouchRipple-root"
          />
        </div>
        <div
          aria-disabled="false"
          class="MuiButtonBase-root MuiListItem-root MuiListItem-dense MuiListItem-gutters MuiListItem-button"
          tabindex="0"
        >
          <div
            class="MuiListItemIcon-root makeStyles-checkbox-5"
          >
            <span
              aria-disabled="false"
              class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-6 MuiCheckbox-root MuiCheckbox-colorPrimary makeStyles-checkbox-5 MuiIconButton-colorPrimary MuiIconButton-edgeStart"
            >
              <span
                class="MuiIconButton-label"
              >
                <input
                  aria-labelledby="checkbox-list-label-Motion Stability"
                  class="PrivateSwitchBase-input-9"
                  data-indeterminate="false"
                  tabindex="-1"
                  type="checkbox"
                  value=""
                />
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                  />
                </svg>
              </span>
            </span>
          </div>
          <div
            class="MuiListItemText-root makeStyles-listItemText-4 MuiListItemText-dense"
            id="checkbox-list-label-Motion Stability"
          >
            Motion Stability
          </div>
          <span
            class="MuiTouchRipple-root"
          />
        </div>
        <div
          aria-disabled="false"
          class="MuiButtonBase-root MuiListItem-root MuiListItem-dense MuiListItem-gutters MuiListItem-button"
          tabindex="0"
        >
          <div
            class="MuiListItemIcon-root makeStyles-checkbox-5"
          >
            <span
              aria-disabled="false"
              class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-6 MuiCheckbox-root MuiCheckbox-colorPrimary makeStyles-checkbox-5 MuiIconButton-colorPrimary MuiIconButton-edgeStart"
            >
              <span
                class="MuiIconButton-label"
              >
                <input
                  aria-labelledby="checkbox-list-label-Energy Spent/Fatigue"
                  class="PrivateSwitchBase-input-9"
                  data-indeterminate="false"
                  tabindex="-1"
                  type="checkbox"
                  value=""
                />
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                  />
                </svg>
              </span>
            </span>
          </div>
          <div
            class="MuiListItemText-root makeStyles-listItemText-4 MuiListItemText-dense"
            id="checkbox-list-label-Energy Spent/Fatigue"
          >
            Energy Spent/Fatigue
          </div>
          <span
            class="MuiTouchRipple-root"
          />
        </div>
      </ul>
      <div
        class="makeStyles-btnGroup-3"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-btn-1"
          data-testid="btn-select-all"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            <span
              aria-disabled="false"
              class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-6 MuiCheckbox-root MuiCheckbox-colorPrimary makeStyles-checkbox-5 MuiIconButton-colorPrimary"
            >
              <span
                class="MuiIconButton-label"
              >
                <input
                  class="PrivateSwitchBase-input-9"
                  data-indeterminate="false"
                  tabindex="-1"
                  type="checkbox"
                  value=""
                />
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                  />
                </svg>
              </span>
            </span>
            Select All
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-btn-1"
          style="margin-left: auto;"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            Done
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
</body>
`;
